packages/flutter/lib/src/widgets/scroll_position.dart

```diff
@@ -466,6 +466,20 @@ class ScrollPosition extends ViewportOffset {
   ScrollActivity get activity => _activity;
   ScrollActivity _activity;
 
+  bool get isScrolling => _activity?.isScrolling ?? false;
+
+  ValueNotifier<bool> _isScrollingNotifier;
```

Comment:
Why not just make this publicly accessible? That way you don't need the add/remove methods.
You also don't need isScrolling itself, since the listener would receive it.